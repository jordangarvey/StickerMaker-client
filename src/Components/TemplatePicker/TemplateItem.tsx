import React, { SFC } from "react";
import { StyleSheet, Text, View } from "react-native";

import Shapes from "./Shapes";


/**
 * Interface for the Template Item props
 */
interface ITemplateItemProps {
	/** The shape of the item */
	shape?: Shapes;
	/** The colour of the item */
	colour?: string;
	/** The text of the sticker */
	text?: string;
}

/**
 * Component to render a Template Item
 */
const TemplateItem: SFC<ITemplateItemProps> = (props) => (
	<View style={[{ backgroundColor: props.colour || "green" }, styles.item]}>
		<Text>{props.text}</Text>
	</View>
);

/**
 * Styles for the Template Item
 */
const styles = StyleSheet.create({
	item: {
		height: 100,
		width: 200
	}
});

export default TemplateItem;
