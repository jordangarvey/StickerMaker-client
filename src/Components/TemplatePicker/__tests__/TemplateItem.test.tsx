import React from "react";
import renderer from "react-test-renderer";

import TemplateItem from "../TemplateItem";


describe("The Template Item componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		const component = renderer.create(
			<TemplateItem/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
