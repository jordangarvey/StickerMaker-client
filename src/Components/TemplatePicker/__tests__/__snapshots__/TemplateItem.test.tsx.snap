// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Template Item componentâ€™s snapshot renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "green",
      },
      Object {
        "height": 100,
        "width": 200,
      },
    ]
  }
>
  <Text />
</View>
`;
